src/common/redis_module/ray_redis_module.cc

```diff
@@ -717,23 +696,76 @@ int TableAppend_RedisCommand(RedisModuleCtx *ctx,
     // necessary since we implement the log with a sorted set, so all entries
     // must be unique, or else we will have gaps in the log.
     RAY_CHECK(flags == REDISMODULE_ZADD_ADDED) << "Appended a duplicate entry";
-    // Publish a message on the requested pubsub channel if necessary.
-    TablePubsub pubsub_channel = ParseTablePubsub(pubsub_channel_str);
-    if (pubsub_channel != TablePubsub::NO_PUBLISH) {
-      // All other pubsub channels write the data back directly onto the
-      // channel.
-      return PublishTableAdd(ctx, pubsub_channel_str, id, data);
-    } else {
-      return RedisModule_ReplyWithSimpleString(ctx, "OK");
-    }
+    return REDISMODULE_OK;
```

Comment:
We may need to be more careful about which functions (node vs. tail) should return a Redis reply and when, to make sure that every Redis command returns exactly one reply. I think it should be:
1. If node func errors, then it should return an error reply.
2. If node func does not error, then it should not return anything.
3. Tail func should always return a Redis reply.

I think this means that for the master commit on [credis](https://github.com/ray-project/credis/blob/master/src/chain_module.h#L352), we are incorrectly handling the case where the node func does not error and the command is executed on a non-tail node. In this case, the node func does not return a reply and the tail func isn't run, so we won't be returning a Redis reply.

Reply:
@stephanie-wang yes, bullets 1-3 are exactly what I have in mind.

For singleton chain case (default deployment), both `ActAsTail()` and `ActAsHead()` will be true.  So if node func does not error, the tail func _will_ be run, and hence we will exactly get 1 reply.

For distributed, >1-chain case, the `ChainReplicate()` interface specifically designed for ray's redis module lacks this distributed support (i.e., propagating the update).  (credis' native `MemberPut_RedisCommand()/Put()` do handle propagation, of course.) . Since it's not handled right now we should make sure 1-node case is correct at the moment.

Reply:
Sounds good, just something to keep in mind once we move to distributed.